dev/WebView2/WebView2.cpp

```diff
@@ -0,0 +1,1932 @@
+            {
+                m_isCoreFailure_BrowserExited_State = true;
+
+                // CoreWebView2 takes care of clearing the event handlers when closing the host,
+                // but we still need to reset the event tokens
+                UnregisterCoreEventHandlers();
+
+                // Null these out so we can't try to use them anymore
+                m_coreWebViewCompositionController = nullptr;
+                m_coreWebViewController = nullptr;
+                m_coreWebView = nullptr;
+                ResetProperties();
+            }
+
+            FireCoreProcessFailedEvent(args);
+        }});
+
+    m_cursorChangedRevoker = m_coreWebViewCompositionController.CursorChanged(winrt::auto_revoke, {
+        [this](auto const& controller, auto const& obj)
+        {
+            m_requestedCursor = controller.Cursor();
+
+            UpdateCoreWindowCursor();
+        }});
+}
+
+winrt::IAsyncAction WebView2::CreateCoreObjects()
+{
+    MUX_ASSERT((m_isImplicitCreationInProgress && !m_isExplicitCreationInProgress) ||
+               (!m_isImplicitCreationInProgress && m_isExplicitCreationInProgress));
+
+    if (m_isClosed)
+    {
+        throw winrt::hresult_error(RO_E_CLOSED, L"Cannot create CoreWebView2 (via Source) for Closed WebView2 element.");
+    }
+    else
+    {
+        m_creationInProgressAsync = std::make_unique<AsyncWebViewOperations>();
+        RegisterXamlHandlers();
+
+        // We are about to attempt re-creation of the environment,
+        // so clear any previous 'Missing Anaheim Warning'
+        Content(nullptr);
+
+        if (!m_isCoreFailure_BrowserExited_State)
+        {
+            // Normally we always need a new environment, the exception being when Anaheim process failed and we get to reuse the existing one
+            co_await CreateCoreEnvironment();
+        }
+
+        if (m_coreWebViewEnvironment)
+        {
+            co_await CreateCoreWebViewFromEnvironment(GetHostHwnd());
+
+            // Do initialization including rendering setup.
+            // Try this now but defer to Loaded event if it hasn't fired yet.
+            TryCompleteInitialization();
+        }
+        else if (m_shouldShowMissingAnaheimWarning)
+        {
+            CreateMissingAnaheimWarning();
+        }
+    }
+
+   co_return;
+}
+
+winrt::IAsyncAction WebView2::CreateCoreEnvironment() noexcept
+{
+    hstring browserInstall;
+    hstring userDataFolder;
+
+    if (!m_options)
+    {
+        // NOTE: To enable Anaheim logging, use: L"--no-sandbox --enable-features=msEmbeddedBrowserVisualHosting --enable-logging=stderr --v=1"
+        m_options = winrt::CoreWebView2EnvironmentOptions();
+        m_options.AdditionalBrowserArguments(L"--enable-features=msEmbeddedBrowserVisualHosting");
+
+        auto applicationLanguagesList = winrt::ApplicationLanguages::Languages();
+        if (applicationLanguagesList.Size() > 0)
+        {
+            m_options.Language(applicationLanguagesList.GetAt(0));
```

Comment:
should this iterate through all of the entries?

Reply:
This is by design. We pass the app's preferred language, obtained by taking the 1st language in the Windows.Globalization.ApplicationLanguages list (note that CWV2Enironment also expects a single language, not a list).